routes/ecom/modules/list-payments.js

```diff
@@ -3,12 +3,40 @@
 // generate default payment gateway object
 const newPaymentGateway = require(process.cwd() + '/lib/new-payment-gateway')
 
+const checkDiscount = (paymentGateway, items) => {
+  const discount = paymentGateway.discount
+  const productIds = discount?.product_ids
```

Comment:
```suggestion
  const productIds = discount && discount.product_ids
```

Node velho, possivelmente sem suporte pra optional chaining